### PR TITLE
Fix NFO parser to be able to read corrct STRM URL format

### PR DESCRIPTION
**Pull request by MaVdbussche**
As per the [Kodi Youtube add-on documentation](https://kodi.wiki/view/Add-on:YouTube), the correct STRM URL format is not the one Jellyfin expects in BaseNfoParser.cs

This PR proposes to support both the deprecated and the new format, restoring the functionality of the "Trailer" button in the web interface and warning the users via a message in the Logger.

**Changes**
- BaseNfoParser.cs has been adapted to support both URL formats;
- BaseNfoParser.cs has been adapted to log a warning in case a deprecated URL is found. That URL will of course still be handled;
- The list of contributors has been adapted, as per the [Contribution guidelines](https://jellyfin.org/docs/general/contributing/development)

**Issues**
Fixes https://github.com/jellyfin/jellyfin/issues/10869

### CODE REVIEW

**Review by Shadowghost — APPROVED** [review]

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 1]
```diff
@@ -460,10 +460,28 @@ protected virtual void FetchDataFromXmlNode(XmlReader reader, MetadataResult<T>
...
-                            StringComparison.OrdinalIgnoreCase));
+                        if (trailer.StartsWith("plugin://plugin.video.youtube/?action=play_video&videoid=", StringComparison.OrdinalIgnoreCase))
+                        {
+                            // Deprecated format
+                            item.AddTrailerUrl(trailer.Replace(
+                                "plugin://plugin.video.youtube/?action=play_video&videoid=",
+                                BaseNfoSaver.YouTubeWatchUrl,
+                                StringComparison.OrdinalIgnoreCase));
+
+                            var suggested_url = trailer.Replace(
```
Comment (crobibero) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:471]:
```suggestion
                            var suggestedUrl = trailer.Replace(
```

We don't use snake_case.

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 2]
```diff
@@ -460,10 +460,28 @@ protected virtual void FetchDataFromXmlNode(XmlReader reader, MetadataResult<T>
...
+                                "plugin://plugin.video.youtube/?action=play_video&videoid=",
+                                "plugin://plugin.video.youtube/play/?video_id=",
+                                StringComparison.OrdinalIgnoreCase);
+                            Logger.LogWarning("Trailer URL uses a deprecated format : {URL}. Using {URL_NEW} instead is advised.", [trailer, suggested_url]);
```
Comment (crobibero) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:475]:
```suggestion
                            Logger.LogWarning("Trailer URL uses a deprecated format : {Url}. Using {NewUrl} instead is advised.", [trailer, suggestedUrl]);
```

**Review by crobibero — CHANGES_REQUESTED** [review]

**Comment by MaVdbussche** [comment]
@crobibero Thank you for the review. Not sure what I was thinking there 😄 
I applied your suggested changes as-is.

**Review by barronpm — APPROVED** [review]

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 3]
```diff
@@ -460,10 +460,28 @@ protected virtual void FetchDataFromXmlNode(XmlReader reader, MetadataResult<T>
...
+                            item.AddTrailerUrl(trailer.Replace(
+                                "plugin://plugin.video.youtube/?action=play_video&videoid=",
+                                BaseNfoSaver.YouTubeWatchUrl,
+                                StringComparison.OrdinalIgnoreCase));
+
+                            var suggestedUrl = trailer.Replace(
+                                "plugin://plugin.video.youtube/?action=play_video&videoid=",
+                                "plugin://plugin.video.youtube/play/?video_id=",
+                                StringComparison.OrdinalIgnoreCase);
+                            Logger.LogWarning("Trailer URL uses a deprecated format : {Url}. Using {NewUrl} instead is advised.", [trailer, suggestedUrl]);
```
Comment (Bond-009) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:475]:
```suggestion
                            Logger.LogWarning("Trailer URL uses a deprecated format : {Url}. Using {NewUrl} instead is advised.", trailer, suggestedUrl);
```

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs (hunk 3 above)
Comment (MaVdbussche) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs]: @Bond-009 Didn't know that was a language feature, makes a lot of sense. Thanks ! It's adapted.

**Review by Bond-009 — APPROVED** [review]

**Comment by MaVdbussche** [comment]
Hello @crobibero , as far as I can tell all requested changes have been addressed. Are you still expecting something from me ?

**Review by crobibero — APPROVED** [review]

**Pull request merged**